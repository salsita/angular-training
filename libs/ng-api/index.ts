export { apiActionTypes, apiActionCreators } from './src/api.actions';
export { apiCall, withLoadingIndicator } from './src/api.decorators';
export { ApiError, BusinessValidationError } from './src/api.errors';
export { API_BASE_URL } from './src/api.interceptor';
export { ApiErrorPayload, ApiModuleConfig, ApiState, ApiStateRoot } from './src/api.interfaces';
export { ApiSelectors } from './src/api.selectors';
export { ApiService } from './src/api.service';
export { HTTP_CONFLICT } from './src/api.status-codes';
export { ApiModule } from './src/api.module';
