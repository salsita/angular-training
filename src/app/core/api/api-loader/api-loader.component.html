<div *ngIf="isLoading$ | async">
  Loading...
</div>
