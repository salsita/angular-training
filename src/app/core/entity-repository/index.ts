export {
  EntityRepositoryModuleConfig,
  EntityRepositoryState,
  EntityRepositoryStateRoot,
  SingleEntityRepository
} from './entity-repository.interfaces';
export { EntityRepositorySelectors } from './entity-repository.selectors';
export { EntityRepositoryService } from './entity-repository.service';
export { EntityRepositoryModule } from './entity-repository.module';
