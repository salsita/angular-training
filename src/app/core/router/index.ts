export {
  FlatRouterStateSnapshot,
  RouterModuleConfig,
  RouterState,
  RouterStateRoot
} from './router.interfaces';
export { RouterSelectors } from './router.selectors';
export { CustomRouterStateSerializer } from './router.serializer';
export { RouterModule } from './router.module';
