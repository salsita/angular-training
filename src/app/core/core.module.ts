import { HttpClientModule } from '@angular/common/http';
import { NgModule, Optional, SkipSelf } from '@angular/core';
import { ApiModule } from '@salsita/ng-api';
import { CrudModule } from '@salsita/ng-crud';
import { EntityRepositoryModule } from '@salsita/ng-entities';
import { RouterModule as RouterStoreModule } from '@salsita/ng-router';

import { environment } from '../../environments/environment';
import { SharedModule } from '../shared/shared.module';
import { ApiErrorToastComponent } from './api/api-error-toast/api-error-toast.component';
import { ApiLoaderComponent } from './api/api-loader/api-loader.component';

@NgModule({
  imports: [
    HttpClientModule,

    ApiModule.forRoot({ storeNamespace: 'api', baseUrl: environment.apiBaseUrl }),
    CrudModule.forRoot({ storeNamespace: 'crud' }),
    EntityRepositoryModule.forRoot({ storeNamespace: 'entityRepository' }),
    RouterStoreModule.forRoot({ storeNamespace: 'router' }),
    SharedModule
  ],
  declarations: [ApiErrorToastComponent, ApiLoaderComponent],
  exports: [ApiErrorToastComponent, ApiLoaderComponent, ApiModule, HttpClientModule]
})
export class CoreModule {
  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        `CoreModule has already been loaded. Import Core modules in the AppModule only.`
      );
    }
  }
}
