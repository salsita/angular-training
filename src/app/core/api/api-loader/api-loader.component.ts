import { Component } from '@angular/core';
import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

import { ApiStateRoot } from '../api.interfaces';
import { isLoading } from '../api.selectors';

@Component({
  selector: 'app-api-loader',
  templateUrl: './api-loader.component.html',
  styleUrls: ['./api-loader.component.scss']
})
export class ApiLoaderComponent {
  isLoading$: Observable<boolean>;

  constructor(store: Store<ApiStateRoot>) {
    this.isLoading$ = isLoading(store);
  }
}
