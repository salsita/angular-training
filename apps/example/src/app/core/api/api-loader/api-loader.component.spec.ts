import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { ApiSelectors } from '@salsita/ng-api';
import { MockApiSelectors } from '@salsita/ng-api/testing';
import { ApiLoaderComponent } from './api-loader.component';

describe('ApiLoaderComponent', () => {
  let fixture: ComponentFixture<ApiLoaderComponent>;
  let selectors: MockApiSelectors;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [ApiLoaderComponent],
        schemas: [NO_ERRORS_SCHEMA],
        providers: [{ provide: ApiSelectors, useClass: MockApiSelectors }]
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(ApiLoaderComponent);
    selectors = TestBed.get(ApiSelectors);
  });

  it('should render disabled loader by', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render enabled loader', () => {
    selectors.subjects.isLoading.next(true);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
