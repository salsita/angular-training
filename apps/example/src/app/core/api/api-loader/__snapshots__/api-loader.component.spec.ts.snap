// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiLoaderComponent should render disabled loader by 1`] = `
<app-api-loader
  isLoading$={[Function Observable]}
>
  
  
</app-api-loader>
`;

exports[`ApiLoaderComponent should render enabled loader 1`] = `
<app-api-loader
  isLoading$={[Function Observable]}
>
  <div>
    
    Loading...
  
  </div>
  
</app-api-loader>
`;
